core/common/src/main/java/alluxio/underfs/UfsStatus.java

```diff
@@ -23,6 +23,7 @@
 @NotThreadSafe
 public abstract class UfsStatus {
   protected final boolean mIsDirectory;
+  protected final Long mLastModifiedTimeMs;
```

Comment:
I suggest using `long` and a value of `-1` for unset. It is tricky for developers to remember to null check a `Long` return type.

Reply:
In general checking last modified time for a file should not worry about null given they are always forced to pass in a non-null value in the constructor. For directories, the last modified value can be null since not all UFSs have modified time for directories.

If we don't want to null check directory timestamps either, Java `File.getLastModified()` returns `0L` to indicate an IO error or if file does not exist. We can do the same here. Not sure if we want to return negative values given that it is not officially supported by the POSIX standard.

Reply:
You should add a comment on what it means if it is `null`.

Reply:
Done.